File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int count = 0;
        int current_nb = 1;
        int i = 0;  // pointer in arr
        int n = arr.size();
        while (i < n) {
            if (arr[i] != current_nb) {
                count++;
                if (count == k) {
                    return current_nb;
                }
            }
            else {
                i++;
            }
            current_nb++;
        }
        return current_nb + (k - count - 1);
    }
};

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.rs
impl Solution {
    pub fn find_kth_positive(arr: Vec<i32>, k: i32) -> i32 {
        let mut count: i32 = 0;
        let mut current_nb: i32 = 1;
        let mut i: usize = 0;  // pointer in arr
        let n: usize = arr.len() as usize;
        while (i < n) {
            if (arr[i] != current_nb) {
                count += 1;
                if (count == k) {
                    return current_nb;
                }
            }
            else {
                i += 1;
            }
            current_nb += 1;
        }
        return current_nb + (k - count - 1);
    }
}